apps/server/src/modules/files-storage/uc/files-storage-delete.uc.spec.ts

```diff
@@ -0,0 +1,333 @@
+import { Counted, EntityId } from '@shared/domain';
+import { AntivirusService } from '@shared/infra/antivirus/antivirus.service';
+import { fileRecordFactory, setupEntities } from '@shared/testing';
+import { Logger } from '@src/core/logger';
+import { AuthorizationService } from '@src/modules/authorization';
+import { S3ClientAdapter } from '../client/s3-client.adapter';
+import { FileRecordParams } from '../controller/dto';
+import { FileRecord, FileRecordParentType } from '../entity';
+import { PermissionContexts } from '../files-storage.const';
+import { FilesStorageMapper } from '../mapper';
+import { FilesStorageService } from '../service/files-storage.service';
+import { FilesStorageUC } from './files-storage.uc';
+
+const getFileRecordsWithParams = () => {
+	const userId = new ObjectId().toHexString();
+	const schoolId = new ObjectId().toHexString();
+
+	const fileRecords = [
+		fileRecordFactory.buildWithId({ parentId: userId, schoolId, name: 'text.txt' }),
+		fileRecordFactory.buildWithId({ parentId: userId, schoolId, name: 'text-two.txt' }),
+		fileRecordFactory.buildWithId({ parentId: userId, schoolId, name: 'text-tree.txt' }),
+	];
+
+	const params: FileRecordParams = {
+		schoolId,
+		parentId: userId,
+		parentType: FileRecordParentType.User,
+	};
+
+	return { params, fileRecords, userId };
+};
+
+describe('FilesStorageUC delete methods', () => {
+	let module: TestingModule;
+	let filesStorageUC: FilesStorageUC;
+	let filesStorageService: DeepMocked<FilesStorageService>;
+	let authorizationService: DeepMocked<AuthorizationService>;
+	let orm: MikroORM;
+
+	const getRequestParams = (schoolId: EntityId, userId: EntityId) => {
+		return { schoolId, parentId: userId, parentType: FileRecordParentType.User };
+	};
+
+	const getParams = () => {
+		const userId: EntityId = new ObjectId().toHexString();
+		const schoolId: EntityId = new ObjectId().toHexString();
+		const requestParams = getRequestParams(schoolId, userId);
+
+		return { userId, schoolId, requestParams };
+	};
+
+	beforeAll(async () => {
+		orm = await setupEntities();
+
+		module = await Test.createTestingModule({
+			providers: [
+				FilesStorageUC,
+				{
+					provide: S3ClientAdapter,
+					useValue: createMock<S3ClientAdapter>(),
+				},
+				{
+					provide: FilesStorageService,
+					useValue: createMock<FilesStorageService>(),
+				},
+				{
+					provide: AntivirusService,
+					useValue: createMock<AntivirusService>(),
+				},
+				{
+					provide: Logger,
+					useValue: createMock<Logger>(),
+				},
+				{
+					provide: AuthorizationService,
+					useValue: createMock<AuthorizationService>(),
+				},
+				{
+					provide: HttpService,
+					useValue: createMock<HttpService>(),
+				},
+			],
+		}).compile();
+
+		filesStorageUC = module.get(FilesStorageUC);
+		authorizationService = module.get(AuthorizationService);
+		filesStorageService = module.get(FilesStorageService);
+	});
+
+	beforeEach(() => {
+		jest.resetAllMocks();
```

Comment:
Do this also reset the golevelup mocks? They should based on jest, but i am not sure if they are also restore over the global jest helper. We only use Once on the most places, but if someone forgett this, a fallback over beforeEach that work for all is very nice.

Reply:
They also reset the golevelup mocks. Thats whoI found some missing mocks in the tests.